### ISSUE TITLE
[Accessibility] Charts: custom show as table

### ISSUE THREAD

**Issue by silvalaura** [cengage/react-magma#2323]
**Reported to Magma:**
1. [On activating ''Show as table'' button, Focus does not move to the close button when the “Tabular representation” dialog opens](https://jira.cengage.com/browse/A11Y-575)
2. [Incorrect use of aria-haspopup and redundant screen reader announcements for "Show as Table" and "View Full Screen" buttons](https://jira.cengage.com/browse/A11Y-610)

**Reported to MT:**

1. [Chart title has heading role but lacks proper heading tag](https://jira.cengage.com/browse/A11Y-4117)
2. [“Show as table" label is not descriptive enough](https://jira.cengage.com/browse/A11Y-4106)
3. ["Show as table" dialog is missing a semantic heading](https://jira.cengage.com/browse/A11Y-4119)
4. [Table heading not semantic](https://jira.cengage.com/browse/A11Y-4180)
5. [Screen reader announces extra content when "Show Table" button receives focus](https://jira.cengage.com/browse/A11Y-4107)
7. [Screen reader remains silent after activate "Show as table"](https://jira.cengage.com/browse/A11Y-4113) 

**Goal**

1. Provide adopters with a way to have a Magma custom show as table modal. In V4 (charts 13), won't be default but v5 (charts 14) we will default to this magma custom solution 
2. This custom modal will display the data similarly to Carbon's solution, but will have proper focus lock, heading semantics and announcements

**Comment by silvalaura** [comment]
Need to look into https://charts.carbondesignsystem.com/api/interfaces/tabularrepcustomizationoptions#tableheadingformatter

<img width="988" height="198" alt="Image" src="https://github.com/user-attachments/assets/e86904fb-3fbb-460e-a0fa-a6f5e9dfc476" />

**Comment by olehnoskov-cengage** [comment]
**Tested on:**
- V4:
Storybook: https://storybook-preview-v4-dev--upbeat-sinoussi-f675aa.netlify.app
Docs: https://next--upbeat-sinoussi-f675aa.netlify.app/data-visualization/chart-toolbar/#overview


**This info needs to be deleted**

<img width="895" height="726" alt="Image" src="https://github.com/user-attachments/assets/3b2f6575-b168-4ae2-a0e3-0a59cb93a95d" />


- V5: 
Storybook: https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/modal--default

Haven't been merged yet.

**Test results:**

[[Accessibility] [Global] [Charts]: On activating ''Show as table'' button, Focus does not move to the close button when the “Tabular representation” dialog opens.](https://jira.cengage.com/browse/A11Y-575) ✅ 

[[Accessibility] [Global] [Charts] [Area (simple)] > Incorrect use of aria-haspopup and redundant screen reader announcements for "Show as Table" and "View Full Screen" buttons](https://jira.cengage.com/browse/A11Y-610) ❌ 

- The "Show as Table" button still uses aria-haspopup="true" instead of dialog.
- `aria-haspopup` hasn't been removed for the 'View Full Screen" button.

[[Accessibility][Keyboard][Bulk Edit][edit student accomodations]: The keyboard focus indicator is not visible when navigating the expanded options in the select dropdown.](https://jira.cengage.com/browse/A11Y-4173)  ❌
- The chart header still uses `<p>` instead of heading tag (e.g., `<h2>`, `<h3>`)

[[Accessibility][JAWS][Activity Performance Report]: "Show as table" label is not descriptive enough](https://jira.cengage.com/browse/A11Y-4106) ✅ 

[[Accessibility][Heading][Activity Performance Report]: "Show as table" dialog is missing a semantic heading
](https://jira.cengage.com/browse/A11Y-4119) ✅ 

[[Accessibility][Certification Prep Performance Report]:Heading is not deined in the 'Table representation' dioalog
](https://jira.cengage.com/browse/A11Y-4180) ✅ 

[[Accessibility][JAWS][Activity Performance Report]: Screen reader announces extra content when "Show Table" button receives focus](https://jira.cengage.com/browse/A11Y-4107)  ✅ 

[[Accessibility][JAWS][Activity Performance Report]: Screen reader remains silent after activate "Show as table"](https://jira.cengage.com/browse/A11Y-4113) ✅

**Comment by olehnoskov-cengage** [comment]
@moathabuhamad-cengage 
Could you cherry-pick this [PR](https://github.com/cengage/react-magma/pull/2382) into the `dev` branch?
